examples/vision/sample_size_estimate.py

```diff
@@ -0,0 +1,545 @@
+            if verbose == 1:
+                print(layer.name, "trainable")
+        else:
+            layer.trainable = False
+            if verbose == 1:
+                print(layer.name, "NOT trainable")
+    print("Trainable weights:", len(model.trainable_weights))
+    print("Non-trainable weights:", len(model.non_trainable_weights))
+    return model
+
+
+"""
+# Define augmentation function
+
+Define image augmentation using keras preprocessing layers.
+"""
+
+image_augmentation = keras.Sequential(
+    [
+        preprocessing.RandomFlip("horizontal"),
+        preprocessing.RandomRotation(0.1),
+        preprocessing.RandomZoom(height_factor=(-0.1, -0), fill_mode="constant"),
+        preprocessing.RandomContrast(factor=0.1),
+    ],
+)
+
+"""
+# Define iterative subsampling functions
+
+To iteratively train a model over several subsample sets we need to create some functions
+for splitting the dataset and training the model.
+
+"""
+
+
+def create_subsets(dataset, fraction):
+    """Creates a subset from a dataset based on a specified fraction. It is
+        divided into training (90%) and validation (10%) sets.
+
+    Arguments:
+        dataset: tf.data.Dataset.
+        fraction: Int, number between 0 and 1.
+
+    Returns:
+        Training and validations datasets with number of subsamples.
+    """
+
+    ds_subset = dataset.take(int(num_train_samples * fraction))
+
+    # Calculate subset length
+    num_subsamples = ds_subset.cardinality().numpy()
+    print("Subset length:", num_subsamples)
+
+    # Split off 90% for training and 10% for validation
+    train_val_split = int(num_subsamples * 0.9)
+    train_ds = ds_subset.take(train_val_split)
+    print("Number of training samples:", train_ds.cardinality().numpy())
+
+    val_ds = ds_subset.skip(train_val_split)
+    print("Number of validation samples:", val_ds.cardinality().numpy())
+
+    # Prepare the training set for training, apply random augmentations
+    train_ds = (
+        train_ds.map(
+            lambda img, lab: (tf.image.resize(img, image_size), lab),
+            num_parallel_calls=AUTO,
+        )
+        .map(
+            lambda img, lab: (img, tf.one_hot(lab, num_classes)),
+            num_parallel_calls=AUTO,
+        )
+        .batch(batch_size)  # First batch images before augmenting
+        .map(
+            lambda img, lab: (image_augmentation(img), lab), num_parallel_calls=AUTO,
+        )  # Augment images
+        .cache()
+        .prefetch(AUTO)
+    )
+
+    # Prepare validation set for training, only resize and one-hot encode
+    val_ds = (
+        val_ds.map(
+            lambda img, lab: (tf.image.resize(img, image_size), lab),
+            num_parallel_calls=AUTO,
+        )
+        .map(
+            lambda img, lab: (img, tf.one_hot(lab, num_classes)),
+            num_parallel_calls=AUTO,
+        )
+        .cache()
+        .batch(batch_size)
+        .prefetch(AUTO)
+    )
+    return train_ds, val_ds, num_subsamples
```

Comment:
The num_subsamples output is used to capture the split sample_sizes when training iteratively. To create the x-values for our plot later. This will fall away when moving to a numpy based approach.